src/integration-test/java/com/mongodb/jdbc/integration/MongoSQLIntegrationTest.java

```diff
@@ -55,4 +77,70 @@ public void loadTestConfigs() throws IOException {
         }
         return dynamicTests;
     }
+
+    @Test
+    public void testTypesOther() throws SQLException, IOException, IllegalAccessException {
+        DataLoader loader = new DataLoader(TEST_DATA_OTHER_DIRECTORY);
+        loader.dropCollections();
+
+        Map<String, Object> row = new LinkedHashMap<>();
+        ObjectId objID = new ObjectId("000000000000000000000001");
+        BsonObjectId objectID = new BsonObjectId(objID);
+        BsonDbPointer dbPointer = new BsonDbPointer("namespace", objID);
+        BsonJavaScript javaScript = new BsonJavaScript("function(){ }");
+        BsonRegularExpression regularExpression = new BsonRegularExpression("a(bc)*");
+        BsonSymbol symbol = new BsonSymbol("symbol");
+        BsonTimestamp timestamp = new BsonTimestamp(100);
+
+        row.put("_id", objectID);
+        row.put("a", dbPointer);
+        row.put("b", javaScript);
+        row.put("c", new BsonMaxKey());
+        row.put("d", new BsonMinKey());
+        row.put("e", regularExpression);
+        row.put("f", symbol);
+        row.put("g", timestamp);
+        /*
+        // Skip Reason - SQL-697
+        // Getting result set schema throwing error: data did not match any variant of untagged enum BsonType
+        BsonArray array = new BsonArray();
+        array.add(new BsonInt32(1));
+        array.add(new BsonInt32(2));
+        array.add(new BsonInt32(3));
+        row.put("h", array);
+         */
+
+        loader.loadTestRow("integration_test", "types_other", row);
```

Comment:
Why aren't we writing this data directly in the yaml file? We may be able to write the data using extended JSON syntax and then have the java driver interpret the extended json info into appropriate bson types.

Reply:
Some of the types were not able to be serialized using the type tags.  BsonObjectID, BsonDbPointer, BsonMaxKey, were some of the types I was having issues with.  Snakeyaml was throwing an exception that either the object could not be casted to a string, or there wasn't a suitable constructor.  I did not try the extended JSON syntax, I will try it out.

Reply:
for extended json syntax, what I'm picturing is roughly:
```
# data.yml
...
    - db: db1
      collection: other_types
      data:
          - { oid: { $oid: "..." }, symbol: { $symbol: "..." }, ... }
...

# dataloader.java
...
    Document d = Document.parse(new Document(row).toJson());
    collection.insertOne(d);
...
```

I haven't tested this yet.

Reply:
Added change to process extended json if the docsJson field is not null.